vernac/vernacentries.ml

```diff
@@ -846,8 +928,10 @@ let vernac_definition_interactive ~atts (discharge, kind) (lid, udecl) bl t =
   let open DefAttributes in
   let scope, local, poly, program_mode, user_warns, typing_flags, using, clearbody =
     atts.scope, atts.locality, atts.polymorphic, atts.program, atts.user_warns, atts.typing_flags, atts.using, atts.clearbody in
-  let canonical_instance, reversible = atts.canonical_instance, atts.reversible in
-  let hook = vernac_definition_hook ~canonical_instance ~local ~poly ~reversible kind in
+  let hook =
+    let canonical_instance, reversible = atts.canonical_instance, atts.reversible in
+    vernac_definition_hook ~atts ~canonical_instance ~local ~poly ~reversible kind
+  in
```

Comment:
I am thinking that it could be better to move more of the parsing of the attributes into `vernac_definition_hook`, but it isn't strictly necessary.